style/infra-style.md

```diff
@@ -0,0 +1,109 @@
+# Infrastructure Team Style Guide Principles
+
+## Why Do We Care
+
+When developers grok each others'* code faster, we can do our work faster.  We read code more often than we write it.
+
+Code readers can be all of these:  the language guru, the developer new to the language, the inexperienced developer, the expert developer, the sysadmin, the product owner ...
+
+It can be our _own_ code, written earlier, that doesn't make sense.
+
+From https://www.techtarget.com/searchsoftwarequality/feature/Follow-Googles-lead-with-programming-style-guides:
+
+When code is consistent and readable across the organization, Shaindel Schwartz advises, engineers see these benefits:
+- ability to focus on what's getting done, rather than the minutiae of how it's coded;
+- quick understanding of any codebase through familiar interfaces and formats;
+- simpler scaling and implementation of automation; and
+- easier change, such as product ownership transfers, new personnel onboarding and code handoffs.
+
+
+## Our Goals:
+- improve maintainability
+- reduce cognitive load when reading code; we grok it faster if choices are consistent across codebases
+- avoid mistakes, e.g. typos
+- continually improve code
+- follow best practices in broader community
+- avoid disagreements about style
+
+- do as much as possible with automated tools
+
+## Tools
+
+Ruby:
+  - [rubocop](https://rubocop.org/) [cop documentation](https://docs.rubocop.org/rubocop) [github](https://github.com/rubocop/rubocop)
+  - [rubocop-rspec cop documentation](https://docs.rubocop.org/rubocop-rspec/cops.html) [github](https://github.com/rubocop/rubocop-rspec)
+  - [rubocop-rails cop documentation](https://docs.rubocop.org/rubocop-rails/cops.html) [github](https://github.com/rubocop/rubocop-rails)
+  - (not often used by us)[rubocop-performance cop documentation](https://docs.rubocop.org/rubocop-performance/cops_performance.html) [github](https://github.com/rubocop/rubocop-performance)
+
+We want these tools to be exercised when a pull request is created, at a minimum.
+
+
+## Editing existing codebases
+
+Most of our work is editing existing code bases, whether they are 3 months old or 15 years old.
+
+### RULE:  Follow the existing rules.  Strive for no new exceptions.
+
+### Should it be a goal to reduce .rubocop_todo.yml to empty?
+
+(TBD)
+
+
+## Starting a New Code Base
+
+Start with default style guides + exceptions below.
+
+
+## When Style exceptions are acceptable
+
+For long classes/methods/blocks, consider SOLID:
+
+    **S**ingle responsibility
+    **O**pen-Closed (open for extension, closed for modification)
+    **L**iskov substitution
+    **I**nterface segregation (clients should not be forced to depend of interfaces they don't use)
+    **D**ependency inversion (depend on abstractions, not concretions)
```

Comment:
Also here, the formatting is throwing me off as it's in a code block when viewed, so we get `**S**` etc, and not bolding/etc.

Reply:
it will be bold when viewed in github;  you are reading the raw text in this PR.

Reply:
No, when I view the file in github it looks the same, it's rendered in a code block:

![Screen Shot 2022-09-19 at 3 42 57 PM](https://user-images.githubusercontent.com/2294288/191132176-a6684744-7407-47a7-8df4-b861f9279313.png)

Reply:
fixed this - thanks for calling it out.